### PR TITLE
db servers restart with views / links

### PR DESCRIPTION
**Pull request by kvahed**
# Pull Request Guidelines

Pull requests are an essential collaborative tool for modern software development. 

The below list is intended to help you figure out whether your code is ready to be reviewed
and merged into ArangoDB. The overarching goal is to:

- Reduce the amount of *recurring* defects
- Reduce the impact to the other developer’s time and energy spent on defects in your code
- Increase the overall autonomy and productivity of individual developers

## Acceptance Checklist

*The below list is not exhaustive, think thoroughly whether the provided information is sufficient.*
*Remove options that do not apply*

### Scope & Purpose

*(Can you describe what functional change your PR is trying to effect?)*

- [ ] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [ ] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [ ] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [ ] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [ ] The behavior change can be verified via automatic tests

#### Related Information

*(Please reference tickets / specification etc )*

- [ ] There is a GitHub Issue reported by a Community User: 
- [ ] There is an internal planning ticket: 
- [ ] There is a *JIRA Ticket number* (In case a customer was affected / involved): 
- [ ] There is a design document: 

### Testing & Verification

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This PR adds tests that were used to verify all changes:

- [ ] Added **Regression Tests** (Only for bug-fixes) 
- [ ] Added new C++ **Unit Tests** (Either GoogleTest or Catch-Test)
   - Did you add tests for a new *RestHandler* subclass ?
   - Did you add new mocks of underlying code layers to be able to verify your functionality ? 
   - ...
- [ ] Added new **integration tests** (i.e. in shell_server / shell_server_aql)
- [ ] Added new **resilience tests** (only if the feature is impacted by failovers)

Additionally:

- [ ] There are tests in an external testing repository (i.e. node-resilience tests, chaos tests)
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

*(Include link to Jenkins run etc)*

> Think about whether the new code you added is modular enough to be
> easily testable by unit tests written with GTest / Catch. It is not good if your feature is so interconnected
> that it prevents other people from writing their own unit gests. It should be possible
> to use your code in future without extensively mocking your classes.
> A bad example that required some extensive effort would be the storage engine API.

### Documentation

> All new Features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the changelog so that
> developers and users have a concise overview. 

- [ ] Added a *Changelog Entry* (referencing the corresponding public or internal issue number)
- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *http API* 
- [ ] Added *Swagger examples* for the http API  

### CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

File: arangod/Cluster/ClusterInfo.cpp [hunk 1]
```diff
@@ -568,6 +568,14 @@ void ClusterInfo::loadPlan() {
   auto start = clock::now();
   auto& agencyCache = clusterFeature.agencyCache();
 
+  // We need to wait for any cluster operation, which needs access to the
+  // agency cache for it to become ready. The essentials in the cluster, namely
+  // ClusterInfo etc, need to start after first poll result from the agency.
+  // This is of great importance to not accidentally delete data facing an
+  // empty agency. There are also other measures that guard against such a
```
Comment (jsteemann) [inline arangod/Cluster/ClusterInfo.cpp:575]:
```suggestion
  // empty agency. There are also other measures that guard against such an
```

File: CHANGELOG [hunk 2]
```diff
@@ -1,7 +1,10 @@
 devel
 -----
 
-* Removed external file loading in a used 3rd party library (Web UI). 
+
+* Fix arangosearch failure when restarting db servers
```
Comment (jsteemann) [inline CHANGELOG:5]:
```suggestion
* Fix `"Plan" is not an object in agency` error messages when restarting DBservers
  that contained arangosearch views with links.
```

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10250/

**Comment by jsteemann** [comment]
It seems that this PR breaks the catch tests on some platforms:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10250/

**Comment by kvahed** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/10265/

File: js/apps/system/_admin/aardvark/APP/frontend/js/views/queryView.js [hunk 3]
```diff
@@ -2054,7 +2054,7 @@
             // fallback
             if (result.fallback && (result.fallback === 'geo' || result.fallback === 'geotable')) {
               $('#geo-switch').addClass('disabled').css('display', 'inline').css('opacity', '0.5');
-              $('#geo-switch').addClass('tippy').attr('title', 'No internet connection. Map is not available.');
+              $('#geo-switch').addClass('tippy').attr('title', 'No internet collection. Map is not available.');
```
Comment (neunhoef) [inline js/apps/system/_admin/aardvark/APP/frontend/js/views/queryView.js:2057]: This seems to be a bad copy/paste error.

Comment (kvahed) [reply js/apps/system/_admin/aardvark/APP/frontend/js/views/queryView.js:2057]: wt...?  how did this change?

File: CHANGELOG [hunk 4]
```diff
@@ -1,6 +1,10 @@
 devel
 -----
 
+* Fix arangosearch failure when restarting db servers
```
Comment (jsteemann) [inline CHANGELOG:4]:
```suggestion
* Fixed `"Plan" is not an object in agency` error messages when restarting DBservers
  that contained arangosearch views with links.
```

File: CHANGELOG [hunk 5]
```diff
@@ -1,6 +1,10 @@
...
+
+* Removed external file loading in a used 3rd party library (Web UI).
+
```
Comment (jsteemann) [inline CHANGELOG:7]:
```suggestion
```

**Pull request merged**